### ISSUE TITLE
[Bug report] Loot value skill seems to be broken.

### ISSUE THREAD

**Issue by Jesper1222** [stoneman2/LethalProgression#27]
When me and my friends played with loot value we would spec all our points into loot value before the map loaded in, and after it had loaded and the value had applied we would respec into other things.
Altough we quickly noticed that the team loot value percentage would not accually match the accual increase in value after doing it a few times.
The values of the items inside would be way higher then the team loot value percentage (Like a axel worth 300 when team loot value is only +40%)
Once we noticed the value was getting way too high for it to be balanced we decided to unspec all out points from value.
However even after unspeccing all our points the loot value still seemed to be unchanged, the loot in the facility would still be worth 300+ even after landing on a diffrent moon.
Seeing as how quickly the loot value seemed to increase in correlation to the skill points spend, and the staying as high as it is after unspeccing.
I think that whenever we unspecced our loot value skill points the value off the loot didnt change at all and still kept the value bonus equal to the points we had previously spend on it, making it so once we respec our points into value they get added ontop of the previous value increase.
Like if we spec for 50% extra loot value, land the ship, unspec, take of, respec 50% into loot value and land again.
Once we land we wouldnt be having a increase of 50% but a increase of 100% instead.
I have a clip of this happing that shows it pretty clearly, but its too big to be attached to the thread tho and im not sure if i can just put 5 links of the chopped up video here
So please tell me if you wanna see the video because we can probs work something out. (Trough discord or something maybe)

Hope this helps!

**Comment by Jesper1222** [comment]
Sorry if its a bit wordy or hard to understand.
Writing long msges explaining stuff isnt my strong suit XD

**Comment by Scrayyg** [comment]
Having a similar issue, I haven't noticed stacking, but the increase is not lining up with the stat value. I'm sitting at a 14.5% increase in loot value and getting items well over double what they normally would be, like a tea kettle worth 200.

**Comment by HennyBicks01** [comment]
i was looking through the code and i think the problem lies in how the lootvalue.cs is set up

```
namespace LethalProgression.Skills
{
    [HarmonyPatch]
    internal class LootValue
    {
        [HarmonyPrefix]
        [HarmonyPatch(typeof(RoundManager), "SpawnScrapInLevel")]
        private static void AddLootValue()
        {
            if (!LP_NetworkManager.xpInstance.skillList.IsSkillListValid())
                return;

            if (!LP_NetworkManager.xpInstance.skillList.IsSkillValid(UpgradeType.Value))
                return;

            RoundManager.Instance.scrapValueMultiplier = RoundManager.Instance.scrapValueMultiplier + (LP_NetworkManager.xpInstance.teamLootValue.Value / 100);
        }

        public static void LootValueUpdate(int change, int newLevel)
        {
            if (!LP_NetworkManager.xpInstance.skillList.IsSkillListValid())
                return;

            if (!LP_NetworkManager.xpInstance.skillList.IsSkillValid(UpgradeType.Value))
                return;

            LP_NetworkManager.xpInstance.TeamLootValueUpdate(change, newLevel);
        }
    }
}
```

everything in here is applying the value as expected but instead of adding that multiplier to the initial value it continuously stacks it on top of the pre-existing loot value bonus included which is why if you allocated 50% to it one round (making the multiplier 1.5) the next round instead of being 1.5 again gets multiplied on top and becomes 2.25. i don't have much expierience with modding but have a good amount with coding and by setting a static variable to hold the initial multiplier you'll be able to scape this issue. some thing like

```
namespace LethalProgression.Skills
{
    [HarmonyPatch]
    internal class LootValue
    {
        private static readonly float BaseScrapValueMultiplier = 1.0f; // Define the base multiplier

        [HarmonyPrefix]
        [HarmonyPatch(typeof(RoundManager), "SpawnScrapInLevel")]
        private static void AddLootValue()
        {
            if (!LP_NetworkManager.xpInstance.skillList.IsSkillListValid())
                return;

            if (!LP_NetworkManager.xpInstance.skillList.IsSkillValid(UpgradeType.Value))
                return;

            // Reset the scrap value multiplier to its base value
            RoundManager.Instance.scrapValueMultiplier = BaseScrapValueMultiplier;

            // Apply the loot value bonus
            RoundManager.Instance.scrapValueMultiplier += (LP_NetworkManager.xpInstance.teamLootValue.Value / 100.0f);
        }

        public static void LootValueUpdate(int change, int newLevel)
        {
            if (!LP_NetworkManager.xpInstance.skillList.IsSkillListValid())
                return;

            if (!LP_NetworkManager.xpInstance.skillList.IsSkillValid(UpgradeType.Value))
                return;

            LP_NetworkManager.xpInstance.TeamLootValueUpdate(change, newLevel);
        }
    }
}
```

and thus will cause you not to have day 6's like

![Screenshot 2023-12-13 213859](https://github.com/stoneman2/LethalProgression/assets/73999671/da0165c8-b29a-4267-8efa-a4c9710c9f5d)

**Comment by stoneman2** [comment]
That's strange, will look into it. Do keep in mind, loot value only applies on map load.

**Comment by cfi2017** [comment]
This is likely due to RoundManager not being recycled over multiple levels. Because you use an additive approach to the multiplier (which is compatible with other multiplier mods), you keep adding to your previously set multiplier.

WeatherMultipliers solves this as follows:
```c#
private static void Prefix(RoundManager __instance)
		{
			//IL_0006: Unknown result type (might be due to invalid IL or missing references)
			//IL_000b: Unknown result type (might be due to invalid IL or missing references)
			//IL_0011: Unknown result type (might be due to invalid IL or missing references)
			//IL_006d: Unknown result type (might be due to invalid IL or missing references)
			//IL_0024: Unknown result type (might be due to invalid IL or missing references)
			//IL_0052: Unknown result type (might be due to invalid IL or missing references)
			LevelWeatherType currentWeather = __instance.currentLevel.currentWeather;
			if (Plugin.ValueMultipliers.ContainsKey(currentWeather))
			{
				float value = Plugin.ValueMultipliers[__instance.currentLevel.currentWeather].Value;
				__instance.scrapValueMultiplier *= value;
				logger.LogInfo((object)$"Set scrap value multiplier ({value}) for current weather \"{currentWeather}\"");
			}
			else
			{
				logger.LogInfo((object)$"No weather multiplier found for \"{currentWeather}\"");
			}
		}

		private static void Postfix(RoundManager __instance)
		{
			//IL_0006: Unknown result type (might be due to invalid IL or missing references)
			//IL_000b: Unknown result type (might be due to invalid IL or missing references)
			//IL_0011: Unknown result type (might be due to invalid IL or missing references)
			//IL_0024: Unknown result type (might be due to invalid IL or missing references)
			LevelWeatherType currentWeather = __instance.currentLevel.currentWeather;
			if (Plugin.ValueMultipliers.ContainsKey(currentWeather))
			{
				float value = Plugin.ValueMultipliers[__instance.currentLevel.currentWeather].Value;
				__instance.scrapValueMultiplier /= value;
				logger.LogInfo((object)$"Scrap generated, resetting scrap value multiplier to its original value of {__instance.scrapValueMultiplier}");
			}
		}
```

You could likely just add a Postfix to revert your change after scrap generation.

**Comment by stoneman2** [comment]
Fixed in 1.4.0

**Comment by cfi2017** [comment]
That fix will be incompatible with mods like WeatherMultipliers as it overrides existing mutlipliers. A better solution would be something like:

```c#
[HarmonyPrefix]
[HarmonyPatch(typeof(RoundManager), "SpawnScrapInLevel")]
private static void AddLootValue()
{
    if (!LP_NetworkManager.xpInstance.skillList.IsSkillListValid())
        return;

    if (!LP_NetworkManager.xpInstance.skillList.IsSkillValid(UpgradeType.Value))
        return;

    float scrapValueAdded = (LP_NetworkManager.xpInstance.teamLootValue.Value / 100);
    // Every time, set it to the default value, then add the multiplier.
    RoundManager.Instance.scrapValueMultiplier = RoundManager.Instance.scrapValueMultiplier + scrapValueAdded;

    LethalPlugin.Log.LogDebug($"Added {scrapValueAdded} to scrap value multiplier, resulting in {RoundManager.Instance.scrapValueMultiplier}");
}

[HarmoyPostfix]
[HarmonyPatch(typeof(RoundManager), "SpawnScrapInLevel")]
private static void RemoveLootValue()
{
    if (!LP_NetworkManager.xpInstance.skillList.IsSkillListValid())
        return;

    if (!LP_NetworkManager.xpInstance.skillList.IsSkillValid(UpgradeType.Value))
        return;

    float scrapValueAdded = (LP_NetworkManager.xpInstance.teamLootValue.Value / 100);
    // Every time, set it to the default value, then add the multiplier.
    RoundManager.Instance.scrapValueMultiplier = RoundManager.Instance.scrapValueMultiplier - scrapValueAdded;

    LethalPlugin.Log.LogDebug($"Removed {scrapValueAdded} to scrap value multiplier, resulting in {RoundManager.Instance.scrapValueMultiplier}");
}
```

**Comment by stoneman2** [comment]
Sure

**Comment by yuequanlingzhi** [comment]
Good job! I also Encountering this bug while playing. Looking forward to the 1.4.0 update
